homeassistant/components/whirlpool/__init__.py

```diff
@@ -48,9 +49,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         backend_selector,
     )
 
-    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    if appliances_manager.washer_dryers or appliances_manager.aircons:
 
-    return True
+        await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+        return True
+
+    _LOGGER.debug("No Appliances found")
+    return False
```

Comment:
Not sure I agree this is the right change, but i don't have a good reason to articulate why... (Aside: It's not explained in the PR description, and perhaps could be separate anyway from adding new functionality anyway...)

Reply:
This is the addition of the sensor platform (isn't it?)
If there are not appliances on the account, why try to load anything?  As we add more platforms, I suppose this would be more generic calling the platforms...

Reply:
There is a test for setup_error where no appliances are found (that is the return False part of this code) - I suppose it's not really an error if no appliances are found in the account.  Maybe there is a better way to declare that no appliances are found?

Reply:
I don't recall ever seeing this behavior before when reviewing other integrations. I don't have strong principles to cite for  why, i just haven't seen it before so it seems odd. I could see having a check like this during config flow setup? Maybe my intuition is that devices may come and go and its ok, not sure

Reply:
Checking at setup in config flow sounds like the right place to do this to me.